core/shared/src/main/scala/zio/internal/impls/LinkedQueue.scala

```diff
@@ -32,32 +44,32 @@ private[zio] final class LinkedQueue[A] extends MutableConcurrentQueue[A] with S
    * performance implications. Having a better solution would be
    * desirable.
    */
-  private[this] val enqueuedCounter = new AtomicLong(0)
-  private[this] val dequeuedCounter = new AtomicLong(0)
+  private[this] val enqueuedCounter = if (addMetrics) new AtomicLong(0) else null
+  private[this] val dequeuedCounter = if (addMetrics) new AtomicLong(0) else null
 
   override def size(): Int = jucConcurrentQueue.size()
 
-  override def enqueuedCount(): Long = enqueuedCounter.get()
+  override def enqueuedCount(): Long = if (addMetrics) enqueuedCounter.get() else 0L
```

Comment:
I guess this is very minor, but in order to avoid adding `addMetrics` as a class field, you could do this instead (and everywhere else I guess):

```suggestion
  override def enqueuedCount(): Long = if (enqueuedCounter ne null) enqueuedCounter.get() else 0L
```

Reply:
Done